From e90df5f6ef13d546723513aebe5c0975632ed3d4 Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Thu, 8 Oct 2015 15:30:25 +0000
Subject: [PATCH 1404/2848] YARN-3780. Should use equals when compare Resource
 in RMNodeImpl#ReconnectNodeTransition.
 Contributed by zhihai xu. (cherry picked from
 commit c7ee6c151c5771043a6de3b8a951cea13f59dd7b)

Conflicts:

	hadoop-yarn-project/CHANGES.txt

(cherry picked from commit aaff9dc39b4af475ff4f28c6ee12ff33f2a50e91)

Change-Id: I1338cbcf510da64dd6b0d88b1b637d5df0bb58c0
---
 .../server/resourcemanager/rmnode/RMNodeImpl.java  |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmnode/RMNodeImpl.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmnode/RMNodeImpl.java
index 777f173..b95bf40 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmnode/RMNodeImpl.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmnode/RMNodeImpl.java
@@ -605,7 +605,8 @@ public void transition(RMNodeImpl rmNode, RMNodeEvent event) {
         rmNode.httpPort = newNode.getHttpPort();
         rmNode.httpAddress = newNode.getHttpAddress();
         boolean isCapabilityChanged = false;
-        if (rmNode.getTotalCapability() != newNode.getTotalCapability()) {
+        if (!rmNode.getTotalCapability().equals(
+            newNode.getTotalCapability())) {
           rmNode.totalCapability = newNode.getTotalCapability();
           isCapabilityChanged = true;
         }
-- 
1.7.9.5

